ooyala_player/ooyala_player.py

```diff
@@ -267,6 +270,32 @@ def student_view_data(self, context=None):
         })
         return data
 
+    @XBlock.json_handler
+    def publish_completion(self, data, dispatch):  # pylint: disable=unused-argument
+        """
+        Entry point for completion for student_view.
+
+        Parameters:
+            data: JSON dict:
+                key: "completion"
+                value: float in range [0.0, 1.0]
+
+            dispatch: Ignored.
+        Return value: If an error occurs, JSON response (200 on success,
+        400 for malformed data) or {"result": 200} for success.
+        """
+        if not data['completion']:
```

Comment:
@macornwell it appears we have some unnecessary statements here. I think we cam simplify this handler like this 
```
value = data.get('completion', None)
if not 0.0 <= value <= 1.0:
    message = u"Invalid completion value {}. Must be in range [0.0, 1.0]"
    raise JsonHandlerError(400, message.format(value)

self.runtime.publish(self, "completion", {"completion": value})
return {"result":"success"}
```

Reply:
I used your code but added back in the following:
`
        if value >= settings.COMPLETION_VIDEO_COMPLETE_PERCENTAGE:
            value = 1.0
`
This was added because the idea is that by getting to that completion point is equal to finishing (1.0).

Reply:
One forwards compatibility note:  `0.0 <= None` will be a TypeError in python 3.  Better to handle the None case explicitly:  `if value is None or not 0.0 <= value <= 1.0:`